lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -123,5 +126,35 @@ def find_latest_rs(deployment_data)
       rs.sync(latest_rs_data)
       rs
     end
+
+    def min_unavailable_rollout
```

Comment:
I'd call this something like `required_rollout_type`, have it always return a value, i.e.:

```ruby
def required_rollout_type
  strategy = @definition.dig('metadata', 'annotations', 'kubernetes-deploy.shopify.io/required-rollout')
  strategy.presence || REQUIRED_ROLLOUT_TYPES.fetch(:full_availability)
end
```

The validation of the annotation value also needs to happen earlier, before the deploy might have run. I'd suggest overwriting `validate_definition` and calling `super` first, then validating required_rollout_type against a list in a constant if that passed.

Reply:
I like Katrina's suggestion above. I think this function as is requires too much knowledge from users (i.e. calling `.blank?` on its result).